### PR TITLE
Dashboard for Wholesalers

### PR DESCRIPTION
**Pull request by paul1029-ife**
Implemented design for the dashboard page in the nibo app.

### CODE REVIEW

File: src/context/ModalContext.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <ModalContext.Provider value={{showModal, setShowModal}}>
+           {children}
+        </ModalContext.Provider>
+      )
+}
+
+export const useModal = () => {
+    const context = useContext(ModalContext);
+    if (!context) {
+      throw new Error("useCart must be used within a CartProvider");
```
Comment (Ehbraheem) [inline src/context/ModalContext.tsx:24]: I'll suggest you make this error text generic.

File: src/components/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
+    // Update state so the next render shows the fallback UI.
+    return { hasError: true, error };
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    // You can log the error to an external service here
+    console.error("Error caught by ErrorBoundary:", error, errorInfo);
```
Comment (Ehbraheem) [inline src/components/ErrorBoundary.tsx:28]: Is this error going to be displayed in the browser console?

File: src/pages/Dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { useModal } from '../context/ModalContext';
+import { AddProductModal } from './Account';
+
+const Dashboard: React.FC = () => {
+    const {showModal, setShowModal} = useModal()
```
Comment (Ehbraheem) [inline src/pages/Dashboard.tsx:6]:
```suggestion
    const {showModal, setShowModal} = useModal(false)
```

Comment (Ehbraheem) [reply src/pages/Dashboard.tsx:6]: This is better for readability purposes.

**Review by Ehbraheem — COMMENTED** [review]
Good job 👍. Just a few suggestions and clarification.

File: src/pages/Dashboard.tsx (hunk 3 above)
Comment (paul1029-ife) [reply src/pages/Dashboard.tsx:6]: Ooh, I'm sorry sir.

File: src/components/ErrorBoundary.tsx (hunk 2 above)
Comment (paul1029-ife) [reply src/components/ErrorBoundary.tsx:28]: Yes sir. It's just a general error boundary for any errors in the application. There's something else that will be rendered on the user's screen when an error is caught in the application. But the console.error will show the error and the description on the console.

File: src/context/ModalContext.tsx (hunk 1 above)
Comment (paul1029-ife) [reply src/context/ModalContext.tsx:24]: It is an error case to handle a situation where a developer accidentally removes the cart context provider from the main application file sir.

Comment (Ehbraheem) [reply src/context/ModalContext.tsx:24]: Yeah, this makes sense for development. I have a feeling it'll be too sensitive expose that kind of information in a customer's browser console.

File: src/components/ErrorBoundary.tsx (hunk 2 above)
Comment (Ehbraheem) [reply src/components/ErrorBoundary.tsx:28]: Same as this comment https://github.com/niboapp/nibo-web/pull/4#discussion_r1864233064

I think sending just **error.message** is good enough. We don't want people to infer the code structure/framework of the site.

File: src/pages/Dashboard.tsx (hunk 3 above)
Comment (Ehbraheem) [reply src/pages/Dashboard.tsx:6]: Yes, that's good but set a default value of what you expect the state variable to hold. It'll be self descriptive and also useful for typings

File: src/context/ModalContext.tsx (hunk 1 above)
Comment (paul1029-ife) [reply src/context/ModalContext.tsx:24]: - [ ] Okay sir. I'd make it more generic. Would something like "CartProvider is not available" be better sir? It stays generic, but also gives future developers an idea of what the error is.

File: src/components/ErrorBoundary.tsx (hunk 2 above)
Comment (paul1029-ife) [reply src/components/ErrorBoundary.tsx:28]: Okay sir. I'll make the necessary changes to it

**Review by Ehbraheem — APPROVED** [review]
LGTM 👍

**Pull request merged**